src/erlcloud_sns.erl

```diff
@@ -276,7 +277,33 @@ list_endpoints_by_platform_application(PlatformApplicationArn, NextToken, Config
 list_endpoints_by_platform_application(PlatformApplicationArn, NextToken, AccessKeyID, SecretAccessKey) ->
     list_endpoints_by_platform_application(PlatformApplicationArn, NextToken, new_config(AccessKeyID, SecretAccessKey)).
 
+-spec list_topics/0 :: () -> [{topics, [[{arn, string()}]]} | {next_token, string()}].
+-spec list_topics/1 :: (undefined | string() | aws_config()) -> [{topics, [[{arn, string()}]]} | {next_token, string()}].
+-spec list_topics/2 :: (undefined | string(), aws_config()) ->  [{topics, [[{arn, string()}]]} | {next_token, string()}].
 
+list_topics() ->
+    list_topics(default_config()).
+
+list_topics(Config) when is_record(Config, aws_config) ->
+    list_topics(undefined, Config);
+list_topics(NextToken) ->
+    list_topics(NextToken, default_config).
+
+list_topics(NextToken, Config) ->
+    Params =
+        case NextToken of
+            undefined -> [];
+            NextToken -> [{"NextToken", NextToken}]
+        end,
+    Doc = sns_xml_request(Config, "ListTopics", Params),
+    Decoded =
+        erlcloud_xml:decode(
+            [{topics, "ListTopicsResult/Topics/member",
+                fun extract_topic_arn/1
+             },
+             {next_token, "ListTopicsResult/NextToken", text}],
+            Doc),
+    Decoded.
```

Comment:
maybe it makes sense to provide a version of this function which iterates over "NextTokens" automatically effectively retrieving all topics without making a client function manually call list_topics(NextToken, Config), there is an example of that in erlcloud_rds